odoo/orm/domains.py

```diff
@@ -1088,6 +1088,22 @@ def _optimize_in_list(condition, _):
     return DomainCondition(condition.field_expr, condition.operator, OrderedSet(value))
 
 
+@operator_optimization(['in', 'not in'])
+def _optimize_in_required(condition, model):
+    """Remove checks against a null value for required fields."""
+    value = condition.value
+    field = condition._field()
+    if (
+        field.falsy_value is None
+        and field.required
+        and (model._table, field.name) in model.env.registry.not_null_columns
```

Comment:
Code is OK.
Just thought of something
new added/modified manual required fields won't update `model.env.registry.not_null_columns` since they only trigger `setup_models` and `init_models` but not `check_null_constraints`